### DISCUSSION TITLE
Listing an Item on Amazon

### DISCUSSION THREAD

**Discussion by ayotade** [abuzuhri/Amazon-SP-API-CSharp#518] [Q&A]
Please help me with the below. I am getting errors with the attributes object.

Thanks
`
            var putlistingsitem = new ParameterPutListingItem();
            putlistingsitem.marketplaceIds = new string[] { "XXXXXXXXXX" };
            putlistingsitem.sellerId = "XXXXXXXXXXX";
            putlistingsitem.sku = "XXXXXXX";
            putlistingsitem.listingsItemPutRequest = new ListingsItemPutRequest() { requirements = Requirements.LISTING_PRODUCT_ONLY, productType = "PET_FEEDER", attributes = new {
                product_name = "Single Meal Automatic Pet Feeder by Petplanet",
                brand_name = "XXX",
                manufacturer = "XXX",
                external_product_id = "88864675108",
                external_product_id_type = "EAN",
                unit_count = 1,
                unit_count_type = "g",
                length_head_to_toe = 7,
                length_head_to_toe_unit_of_measure = "cm",
                length_width_side_to_side = 24,
                length_width_side_to_side_unit_of_measure = "cm",
                length_height_floor_to_top = 13,
                length_height_floor_to_top_unit_of_measure = "cm",
                country_of_origin = "UK",
                standard_price =17.49,
                quantity = 10,
                is_expiration_dated_product =false,
                main_image_url = "https://cvusss.co.uk/image/500x500/99_69417_1528295667_084b6b.jpg"

            }
            };
            var response =amazonConnection.ListingsItem.PutListingsItem(putlistingsitem);`

**Comment by TechyChap** [comment]
Amazon is not very helpful with the attributes in that it just has the data model of "object".
But if you pull the product type API you can see the structure they need for the attributes.

https://developer-docs.amazon.com/sp-api/docs/product-type-api-use-case-guide

For example, country of origin, the JSON looks like this so you'd need to have a suitable object to wrap that.

        "country_of_origin": [
            {
                "value": "CN",
                "marketplace_id": "A1F83G8C2ARO7P"
            }
        ],

For more details, take a look at this discussion - https://github.com/abuzuhri/Amazon-SP-API-CSharp/discussions/132

**Reply by ayotade** [reply]
@TechyChap Thank you for your reply, I really appreciate it. My issue has been resolved by @abuzuhri, Thanks to his advice and assistance.

**Reply by RenzoF** [reply]
hey @ayotade for the future it would be nice if you either add the solution to the problem here or maybe delete the conversation if you no longer need help

**Reply by ayotade** [reply]
> hey @ayotade for the future it would be nice if you either add the solution to the problem here or maybe delete the conversation if you no longer need help

Thank you for your response. I can't find the option to delete the conversation.  Below is the revised final code


            var x = amazonConnection.CatalogItem.GetCatalogItem202204(new FikaAmazonAPI.Parameter.CatalogItems.ParameterGetCatalogItem() { ASIN = "B01I3JW7PK", includedData = new List<IncludedData> { 
                         
                           IncludedData.attributes

                           } });
                        x.Attributes.Remove("street_date");
                     x.Attributes.Remove("variation_theme");
                      x.Attributes.Add("number_of_boxes", new[] { new {
                                value=1

                            }});

            x.Attributes.Add("storage_instructions", new[] { new {
                                value="not applicable"

                            }});
            x.Attributes.Add("country_of_origin", new[] { new {
                                value= "CA"

                            }});

            x.Attributes.Add("use_by_recommendation", new[] { new {
                                value= "not applicable"

                            }});
            x.Attributes.Remove("generic_keyword");

            x.Attributes.Add("product_description", new[] { new {
                                value= "Dry Dog Food"
                            }});

            x.Attributes.Add("condition_type", new[] { new {

                                value ="new_new"
                           }});

            x.Attributes["vat_level"] = new[] { new {

                                value = "full_vat"
            } };
          
           
           
            


          
            
            x.Attributes.Add("fulfillment_availability", new object[] { new { fulfillment_channel_code = "DEFAULT", quantity = 1 }
              });
            x.Attributes.Add("purchasable_offer", new object[] {
                                                    new{
                                                            currency= "GBP",
                                                            

                                                            our_price=new []
                                                                      { new{
                                                                                schedule =new[]
                                                                          { new {
                                                                                    value_with_tax= 105
                                                                                }
                                                                          }}

                                                                      }


                                                         }
                                                  });

            var putlistingsitem = new ParameterPutListingItem();
            putlistingsitem.marketplaceIds = new string[] { "A1F83G8C2ARO7P" };
            putlistingsitem.sellerId = "{SELLER_ID}";

            putlistingsitem.sku = "2857-001";
            putlistingsitem.listingsItemPutRequest = new ListingsItemPutRequest() { requirements = Requirements.LISTING, productType = "PET_FOOD", attributes = x.Attributes };


            var response = amazonConnection.ListingsItem.PutListingsItem(putlistingsitem);

That is my final listing testing and working code thanks to @abuzuhri.
